deeppavlov/pipeline_manager/__init__.py

```diff
@@ -0,0 +1,518 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+from copy import copy, deepcopy
+from datetime import datetime
+from multiprocessing import Pool
+from pathlib import Path
+from shutil import rmtree
+from typing import Union, Dict, Generator, Optional, List
+
+from psutil import cpu_count
+from tqdm import tqdm
+
+from deeppavlov.core.commands.train import read_data_by_config, get_iterator_from_config
+from deeppavlov.core.commands.train import train_evaluate_model_from_config
+from deeppavlov.core.commands.utils import expand_path, parse_config
+from deeppavlov.core.common.errors import ConfigError
+from deeppavlov.core.common.file import read_json
+from deeppavlov.core.common.prints import RedirectedPrints
+from deeppavlov.core.data.data_fitting_iterator import DataFittingIterator
+from deeppavlov.pipeline_manager.observer import Observer
+from deeppavlov.pipeline_manager.pipegen import PipeGen
+from deeppavlov.pipeline_manager.utils import get_num_gpu
+from deeppavlov.pipeline_manager.utils import results_visualization, get_available_gpus
+
+
+def unpack_args(func):
+    """
+    Decorator that unpacks input arguments from tuple or dict.
+    """
+    from functools import wraps
+
+    @wraps(func)
+    def wrapper(args):
+        if isinstance(args, dict):
+            return func(**args)
+        else:
+            return func(*args)
+
+    return wrapper
+
+
+class PipelineManager:
```

Comment:
Don't write your classes in `__init__.py`, write them elsewhere and then just import them in `__init__.py`

Reply:
Ok.